components/connectionmanager/src/lib.rs

```diff
@@ -243,8 +246,30 @@ impl ConnectionManager {
             return;
         }
 
-        let mut futures = Vec::with_capacity(active_inbound_len - self.inbound_limit);
-        for peer in active_inbound.choose_multiple(&mut thread_rng(), active_inbound_len - self.inbound_limit) {
+        let peer_overflow_amount = active_inbound_len - self.inbound_limit;
+
+        let mut futures = Vec::with_capacity(peer_overflow_amount);
+
+        // the following peer selection strategy is designed to ensure that an eclipse attacker most outperform
+        // the best performing peers in any independent metric in order to perform a total eclipse.
+        // while keeping a bias to disconnect from newer peers, and those with concentrated prefix buckets.
+        for peer in EvictionIter::from_peers(&active_inbound)
+            .filter_peers(
+                // We retain a number of peers proportional to the inbound limit.
+                (self.inbound_limit as f64 / RETAIN_RATIO.floor()) as usize,
```

Comment:
`RETAIN_RATIO` is an f64 whose value is `0.4`, I think this was supposed to be like this?

```rust
(self.inbound_limit as f64 * RETAIN_RATIO).floor() as usize,
```

`RETAIN_RATIO.floor()` is always `0`, so this currently is a division by 0.

Reply:
yes, you are correct, it is meant to be outside of brackets, this is a misplacement of the`floor()`